rskj-core/src/main/java/org/ethereum/datasource/KeyValueDataSourceUtils.java

```diff
@@ -104,7 +122,13 @@ public static void validateDbKind(DbKind currentDbKind, String databaseDir, bool
             return;
         }
 
-        DbKind prevDbKind = KeyValueDataSourceUtils.getDbKindValueFromDbKindFile(databaseDir);
+        DbKindValueFromDbKindFileResponse dbKindValueFromDbKindFile  = KeyValueDataSourceUtils.getDbKindValueFromDbKindFile(databaseDir);
+        DbKind prevDbKind = dbKindValueFromDbKindFile.getDbKind();
+
+        if (!dbKindValueFromDbKindFile.doesDbKindFileExists()) {
+            KeyValueDataSourceUtils.generatedDbKindFile(currentDbKind, databaseDir);
```

Comment:
you already generate dbKind file at the line 121. I guess you could simply check dbKind file existence in there. Also probably then you wouldn't need to use that cumbersome `DbKindValueFromDbKindFileResponse` class.

Reply:
Done